mantidproject/mantid
# Fix Crash when showing detectors on a workspace with a missing EFixed

**Description of work.**
This PR fixes a crash when Showing detectors on a workspace which has a spectrum that has no Efixed defined, and is not a monitor.

**To test:**
1. Open mantid and load the data provided in the issue.
2. Right click the workspace and do `Show Detectors`
3. There should be no crash, a detector table should appear.

Fixes #30547

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
